tests/test_init.py

```diff
@@ -127,7 +127,7 @@ def test_results_file_contents_loaded(self, results_file_path, schema_file_path)
                 results_file_path=results_file_path,
                 schema_path=schema_file_path,
             )
-            assert "test_pipe" in psm2.data
+            assert "test_pipe" in psm2.backend.DATA_KEY
```

Comment:
here, and throughout, we're actually using `DATA_KEY` as if it were a constant, when it's actually not a variable...we're actually using the word "DATA_KEY" as the attribute name.

I was the one who did it this way, just because it was convenient as I was testing the implementation, but I'm not sure this is the way we actually want to it to be in the final version...

So I think we should either use just plain: `psm2.backend.data` in all of these places, or we should actually use a CONST like:

```
DATA_KEY = "data"
psm2.backend[DATA_KEY]
```

I think I didn't do it this way because the new backends are not `dict`, so you can't reference an attribute with the item syntax.

Reply:
This is now changed to use self._data in the parent class and self.data in the abstract/child classes.